src/core/main.qml

```diff
@@ -47,121 +47,152 @@ Window {
+                    delayedWelcomeTimer.playWelcome();
+                    delayedbackgroundMusic.start();
+                }
+                else {
+                    DownloadManager.voicesRegistered.connect(
+                                delayedWelcomeTimer.playWelcome);
+                    delayedWelcomeTimer.start();
+                    delayedbackgroundMusic.start();
+                }
             }
         }
     }
-
+    
+    Timer {
+        id: delayedbackgroundMusic
+        interval: 20000
+        repeat: false
+        
+        onTriggered: {
+            print(rccBackgroundMusic)
+            selectedMusicIndex = Math.floor(Math.random()*rccBackgroundMusic.length);
+            backgroundMusic.append(ApplicationInfo.getAudioFilePath("backgroundMusic/"+rccBackgroundMusic[selectedMusicIndex]));
+        }
+    }
+    
     GCAudio {
         id: audioEffects
         muted: !ApplicationSettings.isAudioEffectsEnabled
     }
+    
+    GCAudio {
+        id: backgroundMusic
+        muted: !ApplicationSettings.isBackgroundMusicEnabled
+        
+        onDone: {
+            selectedMusicIndex = Math.floor(Math.random()*rccBackgroundMusic.length);
+            backgroundMusic.append(ApplicationInfo.getAudioFilePath("backgroundMusic/" + rccBackgroundMusic[selectedMusicIndex]));
+        }
+    }
 
     function playIntroVoice(name) {
         name = name.split("/")[0]
         audioVoices.append(ApplicationInfo.getAudioFilePath("voices-$CA/$LOCALE/intro/" + name + ".$CA"))
     }
-
+    
     function checkWordset() {
-        var wordset = ApplicationSettings.wordset
-        if(wordset == '')
-            // Maybe the wordset has been bundled or copied manually
-            // we have to register it if we find it.
-            wordset = 'data2/words/words.rcc'
-
+        if(ApplicationSettings.wordset == '')
+            return
+        
         // check for words.rcc:
         if (DownloadManager.isDataRegistered("words")) {
             // words.rcc is already registered -> nothing to do
-        } else if(DownloadManager.haveLocalResource(wordset)) {
+        } else if(DownloadManager.haveLocalResource(ApplicationSettings.wordset)) {
             // words.rcc is there -> register old file first
             // then try to update in the background
-            if(DownloadManager.updateResource(wordset)) {
-                DownloadManager.resourceRegistered.connect(function() {
-                    // not sure if needed, we check if the resource registered is the one we updated here?
-                    if(wordset === arguments[0]) {
-                        DownloadManager.resourceRegistered.disconnect(arguments.callee);
-                        // force configuration to use the local wordset
-                        ApplicationSettings.wordset = wordset
-                    }
-                })
-            }
-
-
-        } else if(ApplicationSettings.wordset) { // Only if wordset specified
```

Comment:
No changes were required in this function. I will revert them back. I was not sure if that is required to handle it equivalently to wordset here ? Isn't the work done by the Download Manager in the check update that we created? I thought that it would do the job for checking if we already have the .rcc and if not then it would start its job for downloading?